addons/common/functions/fnc_getTargetDistance.sqf

```diff
@@ -14,33 +14,25 @@
  */
 #include "script_component.hpp"
 
-params ["_interval", "_maxDistance", "_minDistance"];
+params [["_accuracy",1], ["_maxDistance",5000], ["_minDistance",0]];
 
-private _position = ATLToASL positionCameraToWorld [0, 0, 0];
-_position set [2, (_position select 2) - (getTerrainHeightASL _position min 0)];
+private _camPosition = AGLToASL positionCameraToWorld [0, 0, 0];
```

Comment:
I know the former code would not have worked over water. The new one probably does, but it should be tested.